core/src/kura.rs

```diff
@@ -413,43 +413,43 @@ impl BlockStore {
     ///
     /// # Panics
     /// * if you pass in `LockStatus::Unlocked` and it is unable to lock the block store.
-    pub fn new(path: &Path, already_locked: LockStatus) -> Self {
+    pub fn new(store_path: &Path, already_locked: LockStatus) -> Self {
         if matches!(already_locked, LockStatus::Unlocked) {
-            let path = path.join(LOCK_FILE_NAME);
+            let lock_path = store_path.join(LOCK_FILE_NAME);
             if let Err(e) = fs::File::options()
                 .read(true)
                 .write(true)
                 .create_new(true)
-                .open(path.clone())
+                .open(lock_path.clone())
             {
                 match e.kind() {
-                    std::io::ErrorKind::AlreadyExists => Err(Error::Locked(path)),
+                    std::io::ErrorKind::AlreadyExists => Err(Error::Locked(lock_path)),
                     std::io::ErrorKind::NotFound => {
-                        match std::fs::create_dir_all(&path)
-                            .map_err(|e| Error::MkDir(e, path.clone()))
+                        match std::fs::create_dir_all(&store_path)
+                            .map_err(|e| Error::MkDir(e, store_path.into()))
```

Comment:
The actual problem was here: `create_dir_all(&path)`, where `path` is the lock file's path.